### PR TITLE
fix for IE 11 tooltip text going out of box

### PR DESCRIPTION
**Pull request by flacoman91**
## Description
In IE, it appears that it doesn't always get the rendered node width.  It looks like there is some sort of lag for the element to get drawn and width calculated.

## Motivation and Context
This fix saves the tooltipRight width and updates it if it is found. 
link to issue: https://github.com/eventbrite/britecharts/issues/650

## How Has This Been Tested?
I made the change and tested it in a virtualbox image running windows 7 and IE 11
the right side text does appear a bit off sometimes, but its better than having it run outside of the box.

The long text titles is still having issues and doesn't wrap all the time.  i think it's the same issue.
https://github.com/eventbrite/britecharts/blob/master/src/charts/tooltip.js#L530

## Screenshots (if appropriate):
sometimes the text appears left or right justified
screen shot 2018-08-06 at 2 22 47 pm
screen shot 2018-08-06 at 2 23 11 pm

## Types of changes

- [ ] Refactor (changes the way we code something without changing its functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] My code follows the [code style guide](https://github.com/eventbrite/britecharts/blob/master/CODESTYLEGUIDE.md) of this project.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by Golodhros** [comment]
Hi @flacoman91, thanks a lot for the fix!

It looks good to me, does it fix the overflowing text we saw in the first screenshot of the issue?

**Comment by flacoman91** [comment]
Ok, here goes.  Using the referenced code in the stackoverflow link that @DalerAsrorov pointed to, i made an adjustment to get the size. 
instead of the long text showing up outside of the box, it looks like the 2nd pic below. I'll keep digging to see if there is something else I can do

screen shot 2018-08-06 at 4 38 11 pm

screen shot 2018-08-06 at 4 38 29 pm

**Comment by DalerAsrorov** [comment]
That is a great effort @flacoman91 - it looks much better than before. Does it still look good in other browsers (Chrome, Firefox)?

**Comment by flacoman91** [comment]
It looks good in the other browsers.

This would be a stopgap solution, and I think that the positioning should be precomputed in a similar manner to how charts are built with the Axis yscale/xscales are done in bar charts.

Our project hasn't gotten to a point yet where we are going to need more than one value in the tooltip nor do we have titles that are that long yet, otherwise I can devote more time to this.

**Comment by Golodhros** [comment]
Thanks a lot @flacoman91!

Thanks for your work on this issue!

We are working on a [rework of the tooltips](https://github.com/eventbrite/britecharts/issues/113), and all these issues will come at hand when actually going over it.

At this moment we are using the canonical text wrapping method that Bostock shared in [this block](https://bl.ocks.org/mbostock/7555321). We know it doesn't work well in IE, but maybe we could use this other project https://github.com/vijithassar/d3-textwrap and see how it works.

File: src/charts/tooltip.js [hunk 1]
```diff
@@ -527,7 +525,17 @@ define(function(require){
...
-                    if (tspan.node().getComputedTextLength() > width) {
+                    // fixes for IE wrap text issue
+                    let temp;
+                    temp = document.createElementNS("http://www.w3.org/2000/svg","text");
+                    temp.appendChild(document.createTextNode(line.join(' ')));
+                    temp.setAttribute('id', 'tempnode');
+                    temp.setAttribute("x",38);
+                    temp.setAttribute("y",18);
+                    document.getElementsByTagName("svg")[0].appendChild(temp);
+                    const textWidth = document.getElementsByTagName("svg")[0].lastChild.getBBox().width;
```
Comment (Golodhros) [inline src/charts/tooltip.js]: I understand we do this to measure the text, right?

We have a small function in our text helper, `getTextWidth`, would this function work?

Comment (flacoman91) [reply src/charts/tooltip.js]: Yes, its to measure the rendered width.  I think the issue is with IE not knowing the size of anything until it's in the DOM.  this link https://stackoverflow.com/a/25168951 shows how to get the correct size in internet explorer

Comment (Golodhros) [reply src/charts/tooltip.js]: I see, I wonder if our method would work in IE11, as it is based in canvas

Comment (flacoman91) [reply src/charts/tooltip.js]: I just changed that to 

```
const textWidth = getTextWidth(line.join(' '), 16, 'Karla, sans-serif');
```

and it does provide the exact same functionality.  You have to pass the font size and font family into the function and it'll work.

Comment (Golodhros) [reply src/charts/tooltip.js]: Awesome!
That function has a default font size and font family, but they are different. Maybe we should update those?

Happy that this worked!

**Comment by flacoman91** [comment]
Ok, i made the change to use the text width helper. 

It still can look like the bunched up text occasionally, but at least this way we're keeping things DRY.  I think it's more performant too since we're not constantly adding and removing things from the DOM.

File: src/charts/tooltip.js [hunk 2]
```diff
@@ -540,8 +543,11 @@ define(function(require){
                                 .text(word);
                         }
                     }
+
+                    // temp.parentNode.removeChild(temp);
```
Comment (Golodhros) [inline src/charts/tooltip.js]: Could we remove this commented code?

Comment (flacoman91) [reply src/charts/tooltip.js]: whoops, that slipped. thanks.

**Comment by Golodhros** [comment]
Thanks a lot for this contribution @flacoman91 !!

**Pull request merged**